### PR TITLE
[train v2][ci] Fix flaky tests and bump up size for tests that are timing out

### LINKED ISSUE #49417: CI test linux://python/ray/train/v2:test_torch_transformers_train is consistently_failing
CI test **linux://python/ray/train/v2:test_torch_transformers_train** is flaky. Recent failures: 
	- https://buildkite.com/ray-project/postmerge/builds/7476#0193f559-0d11-42e5-bdb5-a1165496f14f

DataCaseName-linux://python/ray/train/v2:test_torch_transformers_train-END
Managed by OSS Test Policy

### PR DESCRIPTION
**Pull request by justinvyu**
## Why are these changes needed?

Bumps up the size for flaky tests that are timing out on CI runners.

Also, fix the `test_sync_actor` test to prevent a race condition where the wrong error gets caught by the test assertion.

## Drive-by warning message updates

```
(SynchronizationActor pid=61993) `ray.train.report` has not been called by all 10 workers in the group.
(SynchronizationActor pid=61993) 
(SynchronizationActor pid=61993) The workers have been waiting for ~0.81 s for the following ranks
(SynchronizationActor pid=61993) to join the `report` call: [9].
(SynchronizationActor pid=61993) 
(SynchronizationActor pid=61993) Please ensure that all workers call `ray.train.report` regardless of whether
(SynchronizationActor pid=61993) they participate in checkpointing or not (e.g., pass `checkpoint=None` for ranks
(SynchronizationActor pid=61993) that do not save a checkpoint). Also ensure that workers are not hanging on
(SynchronizationActor pid=61993) other operations, causing them to miss this synchronization barrier.
(SynchronizationActor pid=61993) 
(SynchronizationActor pid=61993) You can set the RAY_TRAIN_REPORT_BARRIER_WARN_INTERVAL_S environment variable to change the frequency
(SynchronizationActor pid=61993) of this warning (current value: 0.2 s).
```

## Related issue number

Closes https://github.com/ray-project/ray/issues/49417

### CODE REVIEW

**Review by raulchen — APPROVED** [review]

**Review by matthewdeng — APPROVED** [review]

File: python/ray/train/v2/tests/test_sync_actor.py [hunk 1]
```diff
@@ -56,37 +56,32 @@ def test_hang():
...
+    mismatch_task = sync_actor.broadcast_from_rank_zero.remote(
+        world_rank=9, world_size=11, data="data-9"
     )
-    # Ensure that the workers are blocked and getting GetTimeoutError after 5 seconds
-    with pytest.raises(ValueError) as excinfo:
-        ray.get(remote_tasks, timeout=1)
-    assert "ValueError" in str(excinfo.type)
+
+    with pytest.raises(ValueError, match="same world size"):
+        ray.get(mismatch_task, timeout=1)
```
Comment (matthewdeng) [inline python/ray/train/v2/tests/test_sync_actor.py:84]: nit: Should this timeout be slightly higher than `timeout_s=1` that the `SynchronizationActor` was initialized with to ensure there isn't a race condition between these two timeouts?

Comment (justinvyu) [reply python/ray/train/v2/tests/test_sync_actor.py:84]: The timeouts here are not needed actually. I'll remove both of them.

The actor who submits the wrong world size will immediately raise the `ValueError`.

**Review by hongpeng-guo — APPROVED** [review]
A side question for this: Are the flaky/ slow tests unique to V2 or are they also exist in V1? If they runs fine in V1, we may need to find the reason leading to the slow down :)

**Comment by justinvyu** [comment]
@hongpeng-guo The tests were medium/large before, but were changed to small in V2.

**Pull request merged**